src/main/java/com/wavefront/sdk/jersey/WavefrontJerseyFilter.java

```diff
@@ -318,4 +386,89 @@ private AtomicInteger getGaugeValue(MetricName metricName) {
       put("jersey" + ".resource.method", finalMethodName);
     }};
   }
+
+  private SpanContext parentSpanContext(ContainerRequestContext requestContext) {
+    Span activeSpan = tracer.activeSpan();
+    if (activeSpan != null) {
+      return activeSpan.context();
+    } else {
+      return tracer.extract(
+              Format.Builtin.HTTP_HEADERS,
+              new ServerHeadersExtractTextMap(requestContext.getHeaders())
+      );
+    }
+  }
+
+  private void decorateRequest(ContainerRequestContext requestContext, Span span) {
+    Tags.COMPONENT.set(span, JERSEY_SERVER_COMPONENT);
+    Tags.HTTP_METHOD.set(span, requestContext.getMethod());
+    String urlStr = null;
+    URL url;
+    try {
+      url = requestContext.getUriInfo().getRequestUri().toURL();
+      urlStr = url.toString();
+    } catch (MalformedURLException e) {
+      // ignoring returning null
+    }
+    if (urlStr != null) {
+      Tags.HTTP_URL.set(span, urlStr);
+    }
+  }
+
+  private void decorateResponse(ContainerResponseContext responseContext, Span span) {
+    int status = responseContext.getStatus();
+    Tags.HTTP_STATUS.set(span, status);
+    if (status >= 400) {
```

Comment:
status >= 400 || status <= 500
convert the logic on line https://github.com/wavefrontHQ/wavefront-jersey-sdk-java/blob/master/src/main/java/com/wavefront/sdk/jersey/WavefrontJerseyFilter.java#L228
into a private util method.

Reply:
we should probably include other 5xx codes too apart from 500.